### PR TITLE
Fixed #26666 -- Added error message when assertRedirects encounters external URLs.

### PR DESCRIPTION
**Pull request by tobiasmcnulty**
See https://code.djangoproject.com/ticket/26666

### CODE REVIEW

File: tests/test_client/tests.py [hunk 1]
```diff
@@ -595,6 +595,14 @@ def test_external_redirect(self):
         response = self.client.get('/django_project_redirect/')
         self.assertRedirects(response, 'https://www.djangoproject.com/', fetch_redirect_response=False)
 
+    def test_external_redirect_with_fetch_error_msg(self):
+        "Regression for https://code.djangoproject.com/ticket/26666"
```
Comment (timgraham) [inline tests/test_client/tests.py]: No need for this comment unless the ticket has some information that's not straightforward.

File: tests/test_client/tests.py [hunk 2]
```diff
@@ -595,6 +595,14 @@ def test_external_redirect(self):
...
+        response = self.client.get('/django_project_redirect/')
+        try:
```
Comment (timgraham) [inline tests/test_client/tests.py]: use `self.assertRaisesMessage`

File: django/test/testcases.py [hunk 3]
```diff
@@ -303,6 +303,11 @@ def assertRedirects(self, response, expected_url, status_code=302,
                 path = urljoin(response.request['PATH_INFO'], path)
 
             if fetch_redirect_response:
+                if netloc:
+                    self.fail(
```
Comment (timgraham) [inline django/test/testcases.py]: I think raising an exception like `ValueError` might be more appropriate.

File: django/test/testcases.py [hunk 4]
```diff
@@ -303,6 +303,11 @@ def assertRedirects(self, response, expected_url, status_code=302,
...
+                        "The Django test client is unable to fetch remote URL %s. "
```
Comment (timgraham) [inline django/test/testcases.py]: ... remote URLs (got %s).

File: django/test/testcases.py [hunk 5]
```diff
@@ -303,6 +303,11 @@ def assertRedirects(self, response, expected_url, status_code=302,
...
+                        "You must set fetch_redirect_response=False to test remote "
```
Comment (timgraham) [inline django/test/testcases.py]: "Use assertRedirects(..., fetch_redirect_response=False) instead."

File: django/test/testcases.py [hunk 6]
```diff
@@ -303,6 +303,11 @@ def assertRedirects(self, response, expected_url, status_code=302,
...
+                        "redirects with assertRedirects." % url)
```
Comment (timgraham) [inline django/test/testcases.py]: put the closing parenthesis on the next line

**Comment by MarkusH** [comment]
Thanks for your work. Merged in c7b1b81352533be9b7f4637c12219f59c0964a82 🔔🔨

**Comment by timgraham** [comment]
Also for future reference, I edited the PR title to try to suggest a more descriptive commit message. Not sure if you tried to keep it short, but we're not too picky about longer messages.

**Pull request closed without merging**